src/query/api/v1/options/handler.go

```diff
@@ -478,3 +507,21 @@ func IsQueryEngineSet(v string) bool {
 	}
 	return false
 }
+
+func (o *handlerOptions) QueryRouter() QueryRouter {
+	return o.queryRouter
+}
+
+func (o *handlerOptions) SetQueryRouter(value QueryRouter) HandlerOptions {
+	o.queryRouter = value
+	return o
```

Comment:
The way we do options is a bit weird; we create a new options while leaving the base immutable to make it safer to chain them together in a style of `opts := opts.SetQueryRouter(foo).SetInstantQuery(bar)` without making permanent changes to the options

 i.e. it'll have to look like:
```golang
func (o *handlerOptions) SetQueryRouter(value QueryRouter) HandlerOptions {
	options := *o
	options.instrumentOpts = opts
	return &options
}
```